### PR TITLE
Introduce global DSP_SHARING configs

### PR DESCRIPTION
**Pull request by nashif**
- kernel: sanitize thread options
- arch: introduce DSP_SHARING and CPU_HAS_DSP configs

### CODE REVIEW

**Comment by nashif** [comment]
huh, not sure what happend here and how I got @aescolar as committer :) Git screwup on my side for sure.

**Review by dcpleung — DISMISSED** [review]

**Review by cfriedt — DISMISSED** [review]
LGTM

File: include/zephyr/kernel.h [hunk 1]
```diff
@@ -214,24 +212,18 @@ void k_thread_foreach_unlocked(
  */
 #define K_CALLBACK_STATE (BIT(4))
 
-#ifdef CONFIG_ARC
-/* ARC processor Bitmask definitions for threads user options */
-
-#if defined(CONFIG_ARC_DSP_SHARING)
```
Comment (ruuddw) [inline include/zephyr/kernel.h:220]: The #if allowed to allocate the scarce option bits differently for different architectures/configured features, but as all options still fit and there is no overlap I guess the #ifdef isn't required.
If needed, the #ifdef can always be added back later.

**Review by ruuddw — DISMISSED** [review]

**Comment by ruuddw** [comment]
Makes sense to lift this to common arch level, I think when it was added for ARC it was requested to keep it local at that time.

**Review by peter-mitsis — DISMISSED** [review]

**Comment by fabiobaltieri** [comment]
@nashif this needs a rebase

**Review by andyross — APPROVED** [review]
Makes sense.  At some point we probably want a general API for "what processor state is part of the context?" questions.  Everything has something like this, c.f. VFPU, NEON, SSE/AVX, HiFi3/4/5 and we aren't consistent about it except for the case of "Well, integer C ABI code is guaranteed to work everywhere"

**Review by ruuddw — APPROVED** [review]

**Pull request merged**